eng/common/scripts/Analyze-Changes.ps1

```diff
@@ -74,6 +74,11 @@ class TreeNode {
 
         # if anything in first level is not 'packages', return true
         foreach ($child in $this.Children) {
+            # skip .prettierignore, .prettierrc.json, cspell.yaml, esling.config.json since these are all covered by github actions globally
```

Comment:
could this just not be in the pipeline definition?

Reply:
maybe?  @hallipr should we exclude in the rules like this or get fancy with our tigger include / excludes?

Reply:
oh, now that I think about it no triggers can't get this fancy.  If you see this test case I added here https://github.com/microsoft/typespec/pull/3344/files#diff-4568c2b31d8896ee70a04b6f0688fa3c580d08ec10e53e71086b5886b884685aR62-R67.  If you modify csharp emitter code and you change cspell.yaml we need to run the csharp portion of pr.yml but we don't need to run the core portion since it didn't change.  We do need to run cspell, but that always runs as a gh action